diff --git a/package/libs/wolfssl/Config.in b/package/libs/wolfssl/Config.in
index b32d5ab6cb..f6fb844a4b 100644
--- a/package/libs/wolfssl/Config.in
+++ b/package/libs/wolfssl/Config.in
@@ -59,6 +59,15 @@ config WOLFSSL_ALT_NAMES
 	bool "Include SAN (Subject Alternative Name) support"
 	default y

+config WOLFSSL_HAS_SP
+	bool "Include Single Precision Math support"
+	default n
+
+config WOLFSSL_HAS_SP_ASM
+	bool "Include the assembly component of the SP math library"
+	depends on WOLFSSL_HAS_SP
+	default n
+
 config WOLFSSL_HAS_DEVCRYPTO
 	bool

@@ -72,6 +81,9 @@ choice
 	config WOLFSSL_HAS_AFALG
 		bool "AF_ALG"

+	config WOLFSSL_HAS_ARMASM
+		bool "ARMv8 Hardware Acceleration"
+
 	config WOLFSSL_HAS_DEVCRYPTO_CBC
 		bool "/dev/crytpo - AES-CBC-only"
 		select WOLFSSL_HAS_DEVCRYPTO
diff --git a/package/libs/wolfssl/Makefile b/package/libs/wolfssl/Makefile
index 15cfd430eb..8957020e64 100644
--- a/package/libs/wolfssl/Makefile
+++ b/package/libs/wolfssl/Makefile
@@ -32,7 +32,8 @@ PKG_CONFIG_DEPENDS:=\
 	CONFIG_WOLFSSL_HAS_ECC25519 CONFIG_WOLFSSL_HAS_OCSP \
 	CONFIG_WOLFSSL_HAS_SESSION_TICKET CONFIG_WOLFSSL_HAS_TLSV10 \
 	CONFIG_WOLFSSL_HAS_TLSV13 CONFIG_WOLFSSL_HAS_WPAS CONFIG_WOLFSSL_HAS_CERTGEN \
-	CONFIG_WOLFSSL_HAS_OPENVPN CONFIG_WOLFSSL_ALT_NAMES
+	CONFIG_WOLFSSL_HAS_OPENVPN CONFIG_WOLFSSL_ALT_NAMES \
+	CONFIG_WOLFSSL_HAS_ARMASM CONFIG_WOLFSSL_HAS_SP CONFIG_WOLFSSL_HAS_SP_ASM

 PKG_ABI_VERSION=$(patsubst %-stable,%,$(PKG_VERSION)).$(call version_abbrev,$(call confvar,$(PKG_CONFIG_DEPENDS)))

@@ -93,7 +94,10 @@ CONFIGURE_ARGS += \
 	--$(if $(CONFIG_WOLFSSL_HAS_DTLS),enable,disable)-dtls \
 	--$(if $(CONFIG_WOLFSSL_HAS_ECC25519),enable,disable)-curve25519 \
 	--$(if $(CONFIG_WOLFSSL_HAS_AFALG),enable,disable)-afalg \
+	--$(if $(CONFIG_WOLFSSL_HAS_ARMASM),enable,disable)-armasm \
 	--$(if $(CONFIG_WOLFSSL_HAS_OPENVPN),enable,disable)-openvpn \
+	--$(if $(CONFIG_WOLFSSL_HAS_SP),enable,disable)-sp \
+	--$(if $(CONFIG_WOLFSSL_HAS_SP_ASM),enable,disable)-sp-asm \
 	--enable-devcrypto=$(if $(CONFIG_WOLFSSL_HAS_DEVCRYPTO_CBC),cbc\
 			  ,$(if $(CONFIG_WOLFSSL_HAS_DEVCRYPTO_AES),aes\
 			  ,$(if $(CONFIG_WOLFSSL_HAS_DEVCRYPTO_FULL),yes,no)))
